docs/src/examples/dual_quaternions.md

```diff
@@ -0,0 +1,157 @@
+
+## Utility functions
+
+First let's load the packages:
+
+```@example dualquat
+using ForwardDiff, Quaternions, Random
+```
+
+Then we'll create some utility types/functions:
+
+```@example dualquat
+const DualQuaternion{T} = Quaternion{ForwardDiff.Dual{Nothing,T,1}}
+
+purequat(p::AbstractVector) = quat(false, @views(p[begin:begin+2])...)
+
+dual(x::Real, v::Real) = ForwardDiff.Dual(x, v)
+
+function dualquat(_q0::Union{Real,Quaternion}, _qe::Union{Real,Quaternion})
+    q0 = quat(_q0)
+    qe = quat(_qe)
+    Quaternion(
+        dual(real(q0), real(qe)),
+        dual.(imag_part(q0), imag_part(qe))...,
+    )
+end
+
+function primal(d::DualQuaternion)
+    return Quaternion(
+        ForwardDiff.value(real(d)),
+        ForwardDiff.value.(imag_part(d))...,
+    )
+end
+
+function tangent(d::DualQuaternion)
+    return Quaternion(
+        ForwardDiff.partials(real(d), 1),
+        ForwardDiff.partials.(imag_part(d), 1)...,
+    )
+end
+
+function dualconj(d::DualQuaternion)
+    de = tangent(d)
+    return dualquat(conj(primal(d)), quat(-real(de), imag_part(de)...))
+end
+
+rotation_part(d::DualQuaternion) = primal(d)
+
+translation_part(d::DualQuaternion) = dualquat(true, conj(rotation_part(d)) * tangent(d))
+
+# first=true returns the translation performed before the rotation: R(p+t)
+# first=false returns the translation performed after the rotation: R(p)+t
+function translation(d::DualQuaternion; first::Bool=true)
+    v = first ? primal(d)' * tangent(d) : tangent(d) * primal(d)'
+    return collect(2 .* imag_part(v))
+end
+
+function transform(d::DualQuaternion, p::AbstractVector)
+    dp = dualquat(true, purequat(p))
+    dpnew = d * dp * dualconj(d)
+    pnew_parts = imag_part(tangent(dpnew))
+    pnew = similar(p, eltype(pnew_parts))
+    pnew .= pnew_parts
+    return pnew
+end
+
+function transformationmatrix(d::DualQuaternion)
+    R = rotationmatrix(rotation_part(d))
+    t = translation(d; first=false)
+    T = similar(R, 4, 4)
+    T[1:3, 1:3] .= R
+    T[1:3, 4] .= t
+    T[4, 1:3] .= 0
+    T[4, 4] = 1
+    return T
+end
+
+randdualquat(rng::AbstractRNG,T=Float64) = dualquat(rand(rng, Quaternion{T}), rand(rng, Quaternion{T}))
+randdualquat(T=Float64) = randdualquat(Random.GLOBAL_RNG,T)
+nothing  # hide
+```
+
+## Example: transforming a point
+
+Now we'll create a dual quaternion.
+```@example dualquat
+x = sign(randdualquat())
+```
+
+`sign(q) == q / abs(q)` both normalizes the primal part of the dual quaternion and makes the tangent part perpendicular to it.
+
+```@example dualquat
+abs(primal(x)) ≈ 1
+isapprox(real(primal(x)' * tangent(x)), 0; atol=1e-10)
+```
+
+Here's how we use dual quaternions to transform a point:
+
+```@example dualquat
+p = randn(3)
+```
+
+```@example dualquat
+transform(x, p)
+```
+
+We can check that this gives the same result as transforming with an affine transformation matrix:
+
+```@example dualquat
+transform(x, p) ≈ (transformationmatrix(x) * vcat(p, 1))[1:3]
+```
+
```

Comment:
````suggestion

## Example: homomorphism from dual quaternions to matrices

```@repl dualquat
y = sign(randdualquat())
```

```@repl dualquat
X = transformationmatrix(x)
Y = transformationmatrix(y)
XY = transformationmatrix(x*y)
X*Y ≈ XY
```
````

I thought it will be better to have a homomorphism example.
If we have this, `y = sign(randdualquat())` definition will be duplicated, and it should be removed in the motion planning section.